eZ/Publish/Core/Event/Tests/SettingServiceTest.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Event\Tests;
+
+use eZ\Publish\API\Repository\Events\Setting\BeforeCreateSettingEvent;
+use eZ\Publish\API\Repository\Events\Setting\BeforeDeleteSettingEvent;
+use eZ\Publish\API\Repository\Events\Setting\BeforeUpdateSettingEvent;
+use eZ\Publish\API\Repository\Events\Setting\CreateSettingEvent;
+use eZ\Publish\API\Repository\Events\Setting\DeleteSettingEvent;
+use eZ\Publish\API\Repository\Events\Setting\UpdateSettingEvent;
+use eZ\Publish\API\Repository\SettingService as SettingServiceInterface;
+use eZ\Publish\API\Repository\Values\Setting\Setting;
+use eZ\Publish\API\Repository\Values\Setting\SettingCreateStruct;
+use eZ\Publish\API\Repository\Values\Setting\SettingUpdateStruct;
+use eZ\Publish\Core\Event\SettingService;
+
+class SettingServiceTest extends AbstractServiceTest
+{
+    public function testUpdateSettingEvents()
```

Comment:
CS: methods need return type
```suggestion
    public function testUpdateSettingEvents(): void
```
applies to all the methods here afaics